manager/integration/tests/test_basic.py

```diff
@@ -600,6 +600,45 @@ def backup_failure_predicate(b):
     cleanup_volume(client, volume)
 
 
+@pytest.mark.skip(reason="TODO")
+def test_backup_block_deletion():  # NOQA
+    """
+    Test backup block deletion
+
+    Context:
+
+    We want to make sure that we only delete non referenced backup blocks,
+    we also don't want to delete blocks while there other backups in progress.
+    The reason for this is that we don't yet know which blocks are required by
+    the in progress backup, so blocks deletion could lead to a faulty backup.
+
+    Setup:
+
+    1. Create a volume and attach to the current node
+    2. Setup NFS backupstore since we can manipulate the content easily
+
+    Steps:
+
+    1.  Write 2 backup blocks of data on to the volume
```

Comment:
Be a little more specific, e.g. 
1. Create and attach the volume on the current node.
2. Write 4 MB to the beginning of the volume (2 x 2MB backup blocks)
3. Create backup1 etc
…
7. Get the backup volume. Check to make sure the volume has 4 blocks.

Reply:
I can make these changes, but to clarify my understanding of test steps description.
Is the objective of the test steps to literally describe the implementation of the python code?

For the block count check one wants to check the volume.cfg as well as the blocks on disk, like I do in the engine integration test. Should I include these details in the descriptions? since they are implementation details?

Reply:
Yes, we should describe the implementation as much as possible, to make sure it's implemented according to the skeleton.

Yes, we can refer to those detail when encountered in the skeleton for the first time, then later the same check operation should follow the same established pattern.